src/pocketmine/Player.php

```diff
@@ -1668,7 +1678,7 @@ public function onUpdate(int $currentTick) : bool{
 			if(!$this->isSpectator() and $this->isAlive()){
 				$this->checkNearEntities($tickDiff);
 
-				if($this->speed !== null){
+				if($this->speed !== null and !$this->isGliding()){
```

Comment:
In this case you would also need anti-cheating checks for elytra speed modification